### DISCUSSION TITLE
Full text search

### DISCUSSION THREAD

**Discussion by tomasweigenast** [isar/isar#202] [Help]
Is Full text search support really full text search? Or the only way is to check for start characters?

**Comment by simc** [comment]
@tomasweigenast 

It really is full text search:

You are right, where clauses (indexes) only support prefix and equality matches. That is the case however for most (all?) databases.
In combination with Isar's support for multi entry indexes you can implement really fast search for almost every use-case.
If you want to use indexes and need more advanced options, I recommend looking into something like the [Porter Stemming Algorithm](https://tartarus.org/martin/PorterStemmer/) or [Soundex](https://en.wikipedia.org/wiki/Soundex). Both work with Isar.

Using filters you also can use `.contains()`, `.endsWith()` and `.matches()`.

Please let me know if you need more explanation or maybe you even want to [contribute to the docs](https://github.com/isar/docs) to make it easier for others to get into full text search.

**Reply by tomasweigenast** [reply]
It is very helpful, thank you. The documentation explains everything really well. I will try to expand it

**Comment by simc** [comment]
@tomasweigenast I created a first draft of a FTS doc. Feel free to extend it https://isar.dev/recipes/full_text_search.html
